### PR TITLE
Remove fakeserver

### PR DESCRIPTION
**Pull request by tpazderka**
- [ ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
- [ ] New code is annotated.
- [ ] Changes are covered by tests.
---
Replace it by testing with responses.

It tends to break on changes to provider and is just basically a separate implementation.

### CODE REVIEW

File: src/oic/oauth2/__init__.py [hunk 1]
```diff
@@ -839,6 +839,7 @@ def do_access_token_request(
             http_args = ht_args
         else:
             http_args.update(ht_args)
+            http_args.pop("password", None)
```
Comment (schlenk) [inline src/oic/oauth2/__init__.py]: why pop password here?

Comment (tpazderka) [reply src/oic/oauth2/__init__.py]: For some reason it was passing through and ended up as an invalid kwarg to requests library...

Comment (schlenk) [reply src/oic/oauth2/__init__.py]: ok. It is wrong in any case, so the fix should do no harm, but probably the issue is burried somewhere in request_info().

Comment (tpazderka) [reply src/oic/oauth2/__init__.py]: Yes, my guess as well. But didn't want to spend a huge amount of time on that. Will make a separate issue for that...

**Pull request merged**